libs/blocks/preflight/panels/assets.js

```diff
@@ -17,7 +21,19 @@ const viewportTooSmall = signal(isViewportTooSmall());
  * Runs asset checks and updates signals with the results.
  */
 async function getResults() {
-  const checks = runChecks(window.location.pathname, document);
+  let checks;
+
+  try {
+    const cachedResults = await executePreflightChecks();
+    if (cachedResults?.assets) {
+      checks = cachedResults.assets;
+    } else {
+      checks = runChecks(window.location.pathname, document);
+    }
+  } catch (error) {
+    console.log('Failed to get cached preflight results, running checks directly:', error);
+    checks = runChecks(window.location.pathname, document);
+  }
```

Comment:
You've duplicated this logic (more than twice...) that's a very good sign it should get centralized somewhere.

We already have the "preflightApi" layer, so ideally, u'd call `preflightApi.getResults`